nixos/modules/services/web-apps/invoiceplane.nix

```diff
@@ -0,0 +1,246 @@
+{ config, pkgs, lib, ... }:
+
+let
+  inherit (lib) mkDefault mkEnableOption mkIf mkMerge mkOption types;
+  inherit (lib) any attrValues flatten literalExpression nameValuePair;
+  inherit (lib) mapAttrs' mapAttrsToList concatStrings;
+
+  cfg = config.services.invoiceplane;
+  eachSite = cfg.sites;
+  user = "invoiceplane";
+  webserver = config.services.${cfg.webserver};
+
+  invoiceplane-config = hostName: cfg: pkgs.writeText "ipconfig.php" ''
+    IP_URL=http://${hostName}
+    ENABLE_DEBUG=false
+    DISABLE_SETUP=false
+    REMOVE_INDEXPHP=false
+    DB_HOSTNAME=${cfg.database.host}
+    DB_USERNAME=${cfg.database.user}
+    DB_PASSWORD=${if cfg.database.passwordFile == null then "" else "file_get_contents(${cfg.database.passwordFile})"}
```

Comment:
`file_get_contents` adds whitespace that you may need to trim. See https://github.com/NixOS/nixpkgs/pull/144250 as an example.

Reply:
Thanks thats a good point!